src/Core/Compiler/CompilerService.cs

```diff
@@ -247,31 +255,31 @@ public IEnumerable<QsNamespaceElement> IdentifyElements(string source)
                     metadata.RoslynMetadatas,
                     options);
 
-                var fromSources = qsCompilation.Namespaces.Select(ns => FilterBySourceFile.Apply(ns, s => s.EndsWith(".qs")));
+                var fromSources = regenerateAll
+                                  ? qsCompilation.Namespaces
+                                  : qsCompilation.Namespaces.Select(ns => FilterBySourceFile.Apply(ns, s => s.EndsWith(".qs")));
 
-                // Only create the serialization if we are compiling for an execution target:
+                // Only create the serialization if we aren't compiling for an execution target:
```

Comment:
Looks like this comment does not even apply now, does it? Looks like we're always doing serialization.

Reply:
Good point, will remove.